manticore/core/cpu/abstractcpu.py

```diff
@@ -762,15 +765,22 @@ def decode_instruction(self, pc):
             c = self.memory[address]
 
             if issymbolic(c):
-                assert isinstance(c, BitVec) and c.size == 8
-                if isinstance(c, Constant):
-                    c = bytes([c.value])
+                # In case of fully symbolic memory, eagerly get a valid ptr
+                if isinstance(self.memory, LazySMemory):
+                    try:
+                        vals = visitors.simplify_array_select(c)
+                        c = bytes([vals[0]])
+                    except visitors.ArraySelectSimplifier.ExpressionNotSimple:
+                        c = struct.pack('B', solver.get_value(self.memory.constraints, c))
                 else:
-                    logger.error('Concretize executable memory %r %r', c, text)
-                    raise ConcretizeMemory(self.memory,
-                                           address=pc,
-                                           size=8 * self.max_instr_width,
-                                           policy='INSTRUCTION')
+                    if isinstance(c, Constant):
```

Comment:
Maybe
```
elif ...:

else:
```

Reply:
This was written this way because previous implementation of this relied on them being separate; currently it's identical. I'll update.